commit-message.md

```diff
@@ -0,0 +1,36 @@
+# Guidelines to create great commit messages
+
+
+We use [GitCop](https://gitcop.com) to check that commit messages are
+properly written. The rules are the following:
+
+* The first line of a commit message, called the subject line should
+  not be more than 80 characters long.
+
+* The commit message should end with the following trailers:
+
+  ```
+  Licence: MIT
+  Signed-off-by: User Name <email@address>
+  ```
+
+  where "User Name" is the author's real name and email@address one of
```

Comment:
On Tue, Jun 09, 2015 at 05:04:11AM -0700, rht wrote:
> > > When it becomes practical it will enable pseudonymous
> > > contributor/reviewer to verify itself.
> >
> > For this purpose, I think it's practical now.
> 
> I think you are referring to 1. pseudonymous contributor/reviewer
> for sign-off or commit author field + pgp-signed emails / fake real
> name, but those are not *signed commits*, 2. one/few author(s) case
> where merges are unneeded/manageable, then sure this is practical
> and has been done for a very long time.

No, I meant author-signed commits that are merged without having the
signature blown away (at least most of the time, see [1]).

> For the second (a) case, so given the choice, you are substituting a
> permanent pki gov id with a pki + one-off 2nd factor auth. Why?

Pseudonymous commits signed by a pseudonymous key allow you to
decouple your government ID from the contributions.  Folks might want
that for all the usual reasons they choose to use pseudonyms.  The PKI
bit just gives you a tool to demonstrate your ownership of the
psuedonym if/when you decide to claim the association.

> So it is up to the lawyers.

I'm pretty sure this is going to be true for copyright issues however
you dice it ;).  And I'm not a lawyer, and I haven't read any previous
case history [2].  But I'd put good odds on a court accepting the
“signed by a pseudonymous key for which I've just demonstrated
secret-key access” argument.

[1]: https://github.com/ipfs/go-ipfs/pull/1338/files#r31879149
[2]: https://github.com/ipfs/go-ipfs/pull/1338/files#r31942822